js/zb.js

```diff
@@ -969,21 +969,82 @@ module.exports = class zb extends Exchange {
     }
 
     async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
-        if (type !== 'limit') {
-            throw new InvalidOrder (this.id + ' allows limit orders only');
-        }
         await this.loadMarkets ();
+        const market = this.market (symbol);
+        const swap = market['swap'];
+        const spot = market['spot'];
+        const timeInForce = this.safeString (params, 'timeInForce');
+        if (type === 'market') {
+            throw new InvalidOrder (this.id + ' createOrder() on ' + market['type'] + ' markets does not allow market orders');
+        }
+        const method = this.getSupportedMapping (market['type'], {
+            'spot': 'spotV1PrivateGetOrder',
+            'swap': 'contractV2PrivatePostTradeOrder',
+        });
         const request = {
-            'price': this.priceToPrecision (symbol, price),
             'amount': this.amountToPrecision (symbol, amount),
-            'tradeType': (side === 'buy') ? '1' : '0',
-            'currency': this.marketId (symbol),
-        };
-        const response = await this.spotV1PrivateGetOrder (this.extend (request, params));
-        return {
-            'info': response,
-            'id': response['id'],
         };
+        if (price) {
+            request['price'] = this.priceToPrecision (symbol, price);
+        }
+        if (spot) {
+            request['tradeType'] = (side === 'buy') ? '1' : '0';
+            request['currency'] = market['id'];
+        } else if (swap) {
+            const reduceOnly = this.safeValue (params, 'reduceOnly');
```

Comment:
Let's not forget to omit `reduceOnly` from `params`, otherwise it will get sent towards the exchange:

```JavaScript
            const reduceOnly = this.safeValue (params, 'reduceOnly');
            params = this.omit (params, 'reduceOnly');
```